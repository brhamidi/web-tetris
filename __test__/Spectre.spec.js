import React from 'react';
import { shallow } from 'enzyme';
import Spectre from '../src/client/components/spectre';

import { List } from 'immutable';

describe('Spectre', () => {
	test('renders correctly in multi', () => {
		const spectreProps = List().set(9, 2).map(e => 0);
		const modeProps = { type: 'multi', meta: { name: "Deneryss" } };
		const wrapper = shallow(<Spectre mode={modeProps} spectre={spectreProps} />);

		expect(wrapper).toMatchSnapshot();
	});
	test('renders correctly in solo', () => {
		const spectreProps = List().set(9, 2).map(e => 0);
		const modeProps = { type: 'solo', meta: { name: undefined } };
		const wrapper = shallow(<Spectre mode={modeProps} spectre={spectreProps} />);

		expect(wrapper).toMatchSnapshot();
	});
});
