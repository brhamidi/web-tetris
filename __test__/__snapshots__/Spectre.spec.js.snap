// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spectre renders correctly in multi 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "display": "flex",
      "flexFlow": "row wrap",
      "justifyContent": "center",
      "width": "25%",
    }
  }
>
  <h2
    style={
      Object {
        "color": "white",
      }
    }
  >
    [
    Deneryss
    ]
  </h2>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "border": "solid white 3px",
        "display": "flex",
        "height": "450px",
        "width": "90%",
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="1"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="2"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="3"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="4"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="5"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="6"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="7"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="8"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
    <div
      key="9"
      style={
        Object {
          "backgroundColor": "white",
          "height": "0%",
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`Spectre renders correctly in solo 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "display": "flex",
      "flexFlow": "row wrap",
      "justifyContent": "center",
      "width": "25%",
    }
  }
>
  <h2
    style={
      Object {
        "color": "white",
      }
    }
  >
    Solo Mode
  </h2>
</div>
`;
